test/check-application

```diff
@@ -944,6 +948,261 @@ class TestFiles(testlib.MachineCase):
         b.wait_visible("[data-item='newfile2']")
         b.click(".breadcrumb-button:nth-of-type(3)")
 
+    @testlib.skipBrowser(".upload_files() doesn't work on Firefox", "firefox")
+    @testlib.skipImage("Debian-testing has Cockpit 311 without new upload support in fsreplace1", "debian-testing")
+    def testUpload(self):
+        b = self.browser
+        m = self.machine
+
+        self.enter_files()
+
+        def get_piechart_progress(sel):
+            b.wait_visible(sel)
+            style = b.attr(sel, "style")
+            m = re.search(r"--progress: (\d+).\d+%;", style)
+            assert m is not None
+            return int(m.group(1))
+
+        def dir_file_count(directory):
+            return m.execute(f"find {directory} -mindepth 1 -maxdepth 1 | wc -l")
+
+        with tempfile.TemporaryDirectory() as tmpdir:
+            # Test cancelling of upload
+            big_file = str(Path(tmpdir) / "bigfile.img")
+            subprocess.check_call(["truncate", "-s", "1500MB", big_file])
+
+            m.execute("runuser -u admin mkdir /home/admin/Downloads")
+            b.wait_visible("[data-item='Downloads']")
+            b.mouse("[data-item='Downloads']", "dblclick")
+            b.wait_not_present(".pf-v5-c-empty-state")
+
+            file_count_before = dir_file_count("/home/admin/Downloads")
```

Comment:
That should be predictably "0" now, no? (That was the whole point of creating a new dir)

Reply:
Ahhh that makes sense, I made it count before and after because well hmm I thought otherwise it might be racy. But in reality this should never happen so adjusted both assertions.

Reply:
Cheers!